PMMain/Midas/MidasGUI/MidasThread.cpp

```diff
@@ -41,3 +42,97 @@ void MidasThread::threadEmitStateString(std::string str)
 {
     emit emitStateString(str);
 }
+
+
+std::vector<ringData> MidasThread::readKeyboardSetupFile()
+{
+    
+    std::ifstream readRingData("C:/Users/Owen/Documents/keyboardData.txt");
+    std::string temp;
+    int vect_num = 0;
+    int ringInOutSwitch = 0; //0 = ringIn , 1 = ringOut 
```

Comment:
If 0 is supposed to represent ringIn and 1 is supposed to represent ringOut, you should define constants or create an enum instead of writing the intended meaning in a comment. It's a lot clearer to a reader of the code.